contracts/tokens/SecurityToken.sol

```diff
@@ -485,7 +479,8 @@ contract SecurityToken is ERC20, ERC20Detailed, Ownable, ReentrancyGuard, Securi
      * @notice Permanently freeze issuance of this security token.
      * @dev It MUST NOT be possible to increase `totalSuppy` after this function is called.
      */
-    function freezeIssuance() external isIssuanceAllowed isEnabled("freezeIssuanceAllowed") onlyOwner {
+    function freezeIssuance(bytes calldata _signature) external isIssuanceAllowed onlyOwner {
+        require(owner() == TokenLib.recoverFreezeIssuanceAckSigner(_signature), "Owner did not sign");
```

Comment:
the signer is always the owner? and msg.sender is also the owner. is it right or I am missing something here

Reply:
yes, the signer is the owner and the msg.sender. This is just an additional acknowledgement from the owner that these actions are irrevocable. Refer to asana task for details.